apps/main/src/dex/components/PagePool/PoolDetails/PoolStats/index.tsx

```diff
@@ -23,9 +24,10 @@ type PoolStatsProps = {
 
 const PoolStats = ({ curve, routerParams, poolAlert, poolData, poolDataCacheOrApi, tokensMapper }: PoolStatsProps) => {
   const tokenAlert = useTokenAlert(poolData?.tokenAddressesAll ?? [])
-  const { rChainId, rPoolId } = routerParams
-  const rewardsApy = useStore((state) => state.pools.rewardsApyMapper[rChainId]?.[rPoolId])
-  const tvl = useStore((state) => state.pools.tvlMapper[rChainId]?.[rPoolId])
+  const { rChainId, rPoolIdOrAddress } = routerParams
+  const poolId = usePoolIdByAddressOrId({ chainId: rChainId, poolIdOrAddress: rPoolIdOrAddress })
+  const rewardsApy = useStore((state) => state.pools.rewardsApyMapper[rChainId]?.[poolId ?? ''])
```

Comment:
You'll find `poolId ?? ''` a few times. I preferred this over putting the `?? ''` at the end of `usePoolIdByAddressOrId`.

This is a temporary annoying null check until we refactor the pool zustack stores to tanstack queries. Array indexers can't accept `undefined`. It's just a typescript thing, because `someArray[undefined]` and `someArray['']` both return `undefined`, so it behaves like it always has.